### PR TITLE
Fix eye movement frame indices

### PR DESCRIPTION
**Pull request by romanroibu**
Currently, the eye movement segment's `start_frame_index` and `end_frame_index` properties point inside the input buffer for `nslr_hmm.classify_gaze`. This PR updates these properties to represent indices into the capture video frame buffer.

### CODE REVIEW

File: pupil_src/shared_modules/eye_movement/model/segment.py [hunk 1]
```diff
@@ -443,12 +451,18 @@ def create_segment(
         segment_class = Segment_Class.from_nslr_class(nslr_segment_class)
         topic = utils.EYE_MOVEMENT_TOPIC_PREFIX + segment_class.value
 
-        start_frame_index, end_frame_index = nslr_segment.i  # [i_0, i_1)
         start_frame_timestamp, end_frame_timestamp = (
             segment_time[0],
             segment_time[-1],
         )  # [t_0, t_1]
 
+        if isinstance(capture_time, np.ndarray) and capture_time.size > 1:
```
[the remark this answers is not in the document]
Comment (romanroibu) [reply pupil_src/shared_modules/eye_movement/model/segment.py]: This was initially `len(capture_time) > 1` but I was hitting weird cases when even though `capture_time` was reported to be an `np.ndarray` instance, calling `len(capture_time)` was raising an exception. I couldn't find the root cause of this, but the check that I implemented here seemed to work every time.

[the remark this answers is not in the document]
Comment (romanroibu) [reply pupil_src/shared_modules/eye_movement/model/segment.py]: nvm, found the cause, fixed it and applied the suggestion

File: pupil_src/shared_modules/eye_movement/worker/offline_detection_task.py [hunk 2]
```diff
@@ -90,6 +90,7 @@ def serialized_dict(datum):
             use_pupil=use_pupil,
             nslr_segment=nslr_segment,
             nslr_segment_class=nslr_segment_class,
+            capture_time=capture.timestamps,
```
Comment (papr) [inline pupil_src/shared_modules/eye_movement/worker/offline_detection_task.py]:
```suggestion
            world_timestamps=capture.timestamps,
```

File: pupil_src/shared_modules/eye_movement/worker/real_time_buffered_detector.py [hunk 3]
```diff
@@ -108,6 +108,7 @@ def _segment_generator(
                 use_pupil=use_pupil,
                 nslr_segment=nslr_segment,
                 nslr_segment_class=nslr_segment_class,
+                capture_time=capture.timestamps,
```
Comment (papr) [inline pupil_src/shared_modules/eye_movement/worker/real_time_buffered_detector.py]:
```suggestion
                world_timestamps=capture.timestamps,
```

**Pull request merged**